libcaja-private/caja-directory.h

```diff
@@ -53,18 +56,12 @@
 #define CAJA_DIRECTORY_GET_CLASS(obj) \
   (G_TYPE_INSTANCE_GET_CLASS ((obj), CAJA_TYPE_DIRECTORY, CajaDirectoryClass))
 
-/* CajaFile is defined both here and in caja-file.h. */
-#ifndef CAJA_FILE_DEFINED
-#define CAJA_FILE_DEFINED
-typedef struct CajaFile CajaFile;
-#endif
```

Comment:
was there a reason for this?  otherwise, it's better indeed, but make sure *caja-file.h* doesn't have conditional dependencies on this header

Reply:
This is good, but I think it would be better to use another PR to modify this. So, I will revert it in this PR.